aea/crypto/ledger_apis.py

```diff
@@ -188,6 +189,63 @@ def is_tx_settled(self, identifier: str, tx_digest: str) -> bool:
             self._last_tx_statuses[identifier] = ERROR
         return is_successful
 
+    def is_tx_valid(
+        self,
+        identifier: str,
+        tx_digest: str,
+        seller: Address,
+        client: Address,
+        tx_nonce: str,
+        amount: int,
+    ) -> bool:
+        """
+        Check whether the transaction is valid
+
+        :param identifier: Ledger identifier
+        :param tx_digest:  the transaction digest
+        :param seller: the address of the seller.
+        :param client: the address of the client.
+        :param tx_nonce: the transaction nonce.
+        :param amount: the amount we expect to get from the transaction.
+        :return: True if is valid , False otherwise
+        """
+        assert identifier in self.apis.keys()
+        is_settled = self._is_tx_settled(identifier=identifier, tx_digest=tx_digest)
```

Comment:
Why do we need that? Can the tx_settled and tx_valid not be merged? I think there should be one function called `is_tx_settled_and_valid`. You really want to know both things and there should be only one call to the blockchain, not two.

Reply:
or `is_valid_settled_tx` might be more succinct.